tests/testthat/test-renderer2-widerect.R

```diff
@@ -35,83 +48,90 @@ test_that("bottom of widerect is above line", {
-                showSelected="year", clickSelects="country",
-                data=not.na, stat="identity", position="identity")+
-       coord_flip(),
-
-       time=list(variable="year", ms=2000),
-       duration=list(year=2000),
-       first=list(year=1975, country=c("United States", "Vietnam")),
-       selector.types=list(country="multiple"),
-       title="World Bank data (multiple selection, facets)")
+wb.facets <- animint(
+  ts=ggplot()+
+    xlab("")+
+    geom_tallrect(aes(
+      xmin=year-1/2, xmax=year+1/2,
+      linetype=status),
+      clickSelects="year",
+      data=TS(years), alpha=1/2)+
+    theme_bw()+
+    theme_animint(width=1000, height=800)+
+    theme(panel.margin=grid::unit(0, "lines"))+
+    geom_line(aes(
+      year, life.expectancy, group=country, colour=region,
+      id = country),
+      clickSelects="country",
+      data=TS(not.na), size=4, alpha=3/5)+
+    geom_point(aes(
+      year, life.expectancy, color=region, size=population),
+      clickSelects="country",
+      showSelected="country",
+      data=TS(not.na))+
+  geom_path(aes(
+    fertility.rate, year, group=country, colour=region),
+    clickSelects="country",
+    data=TS2(not.na), size=4, alpha=3/5)+
+  geom_point(aes(
+    fertility.rate, year, color=region, size=population),
+    showSelected="country", clickSelects="country",
+    data=TS2(not.na))+
+  geom_widerect(aes(
+    ymin=year-1/2, ymax=year+1/2,
+    linetype=status,
+    id=paste0("year", year)),
+    clickSelects="year",
+    data=TS2(years), alpha=1/2)+
+  geom_point(aes(
+    fertility.rate, life.expectancy,
+    colour=region, size=population,
+    key=country), # key aesthetic for animated transitions!
+    clickSelects="country",
+    showSelected="year",
+    data=SCATTER(not.na))+
+  geom_text(aes(
+    fertility.rate, life.expectancy, label=country,
+    key=country), #also use key here!
+    showSelected=c("country", "year"),
+    clickSelects="country",
+    data=SCATTER(not.na))+
+  scale_size_animint(breaks=10^(5:9))+
+  facet_grid(side ~ top, scales="free")+
+  geom_text(aes(
+    5, 85, label=paste0("year = ", year), 
+    key=year),
+    showSelected="year",
+    data=SCATTER(years)),
+  bar=ggplot()+
+    theme_animint(height=2400)+
+    geom_bar(aes(
+      country, life.expectancy, fill=region,
+      key=country, id=gsub(" ", "_", country)),
+      showSelected="year", clickSelects="country",
+      data=not.na, stat="identity", position="identity")+
+    coord_flip(),
+  time=list(variable="year", ms=2000),
+  duration=list(year=2000),
+  first=list(year=1975, country=c("United States", "Vietnam")),
+  selector.types=list(country="multiple"),
+  title="World Bank data (multiple selection, facets)",
+  source="https://github.com/animint/animint2/blob/master/tests/testthat/test-renderer2-widerect.R")
 
 info <- animint2HTML(wb.facets)
+expect_source("https://github.com/animint/animint2/blob/master/tests/testthat/test-renderer2-widerect.R")

```

Comment:
this test checks that there is a `<a>` tag created, when the user viz has source option.